File: api/src/models/event.js
const mongoose = require("mongoose");

const { Schema } = mongoose;

const eventSchema = new Schema(
  {
    title: {
      trim: true,
      required: true,
      type: String,
    },
    description: {
      required: true,
      type: String,
    },
    dateEvent: {
      trim: true,
      required: true,
      type: Date,
    },
    hasPassed: {
      required: false,
      type: Boolean,
      default: false,
    },
    template: {
      type: mongoose.Schema.Types.ObjectId,
      required: false,
      ref: "Template",
    },
  },
  {
    timestamps: true,
  },
);

const Event = mongoose.model("Event", eventSchema);
module.exports = Event;

File: client/lib/models/event.dart
import 'package:client/models/template.dart';

class Event {
  String? title;
  String? description;
  DateTime? dateEvent;
  bool? hasPassed;
  Template? template;
  Event(
      {this.title,
      this.description,
      this.dateEvent,
      this.hasPassed,
      this.template});
  Event.fromMap(Map map)
      : this(
            title: map['title'],
            description: map['description'],
            dateEvent: map['dateEvent'],
            hasPassed: map['hasPassed'],
            template: map['template']);
  Map<String, dynamic> asMap() => {
        'title': title,
        'description': description,
        'dateEvent': dateEvent,
        'hasPassed': hasPassed,
        'template': template
      };
  static notNull(dynamic element) => element != null;
}
